Python/wrfrun/pyWPS.py

```diff
@@ -264,6 +263,148 @@ class ERAI(Dataset):
+  def cleanup(self, tgt):
+    cwd = os.getcwd()
+    os.chdir(tgt)
+    os.remove(self.ungrib_exe)
+    os.remove(self.fixIM)
+    os.remove(self.vtable)
+    os.chdir(cwd)
+
+  # ======= Method to determine whether a subfolder contains valid data and can be processed =======
+  # ======= recursively. Checks that the subfolder name is a valid calendar year.            =======
+  def checkSubDir(self, subdir, start, end):     
+    match = self.subdregx.match(subdir)
+    if match:      
+      lmatch = ( start[0] <= int(subdir) <= end[0] )
+    else: lmatch = False 
+    return lmatch 
+
+  # ================= Method that generates the WRF IM file for metgrid.exe ===============
+  def ungrib(self, date, mytag):  
+    # Create formatted date string
+    datestr = self.datestr.format(*date) # (years, months, days, hours).
+    # Initilize status output message (message printed later)
+    msg = datestr+", with files:"    
+    # Create links to relevant source data
+    Grbfiles = [] # List of relevant source files.    
+    for GrbDir,grbstr in zip(self.GrbDirs,self.grbstrs):
+      # NOTE: The zip() function returns a zip object, which is an iterator of tuples 
+      #   where the first item in each passed iterator is paired together, and then 
+      #   the second item in each passed iterator are paired together, etc.
+      grbfile = grbstr.format(datestr+'00') # Insert current date.
+      Grbfile = '{0:s}/{1:04d}/{2:s}'.format(GrbDir,date[0],grbfile) # Absolute path.
+      if not os.path.exists(Grbfile): 
+        raise IOError("Input file '{0:s}' not found!".format(Grbfile))     
+      else:
+        msg += '\n     '+grbfile # Add to output message.
+        Grbfiles.append(Grbfile) # Append to file list.
+    # Prompt on screen
+    print(('\n   '+mytag+' Processing time-step: '+msg))
+    # Link    
+    for Gribfile,gribname in zip(Grbfiles,self.gribnames): os.symlink(Gribfile,gribname)
+    # NOTE: The format for os.symlink is os.symlink(src, dst).       
+    # Prompt on screen
+    print('\n   * '+mytag+' Converting grib to WRF IM format (ungrib.exe).')    
+    # Run ungrib.exe
+    fungrib = open(self.ungrib_log, 'a') # ungrib.exe output and error log.
+    # NOTE: "a" above means append - opens a file for appending, creates 
+    #   the file if it does not exist. 
+    subprocess.call([self.UNGRIB], stdout=fungrib, stderr=fungrib)
+    # NOTE: The subprocess module allows you to spawn new processes, connect to their 
+    #   input/output/error pipes, and obtain their return codes. With subprocess.call() 
+    #   you pass an array of commands and parameters. This expects input command and
+    #   its parameters inside [].
+    fungrib.close() # Close log file for ungrib.    
+    # Prompt on screen
+    print('\n   * '+mytag+' Fixing WRF IM file (fixIM.py).')    
+    # Run fixIM.py
+    ffixim = open(self.fixIM_log, 'a') # fixIM.py output and error log.
+    call_arg = self.FIXIM   
```

Comment:
not sure why you have this... you can just put self.FIXIM below...

Reply:
Done.